addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsOptionsPanel.java

```diff
@@ -46,15 +49,33 @@ public CustomPayloadsOptionsPanel() {
         gbc.anchor = GridBagConstraints.LINE_START;
         gbc.fill = GridBagConstraints.BOTH;
 
-        this.add(new JLabel(OPTIONS_TITLE), gbc);
+        this.add(getTitleLabel(), gbc);
         gbc.weighty = 1.0;
         this.add(tablePanel, gbc);
     }
 
+    private static JLabel getTitleLabel() {
+        if (titleLabel == null) {
+            titleLabel = new JLabel(OPTIONS_TITLE);
+        }
+        return titleLabel;
+    }
+
+    private static void setTitleLabel(String title) {
+        getTitleLabel().setText(title);
+    }
+
     @Override
     public void initParam(Object obj) {
         OptionsParam optionsParam = (OptionsParam) obj;
         CustomPayloadsParam param = optionsParam.getParamSet(CustomPayloadsParam.class);
+        if (param.getCategoriesNames().isEmpty()) {
+            this.setEnabled(false);
```

Comment:
`tablePanel.setComponentEnabled(false);` and similar for true. Note that you also need to override that method to disable custom buttons.